### ISSUE TITLE
Load dlls from buffer

### ISSUE THREAD

**Issue by cunev** [Koromix/luigi#9]
Hi, would it be possible to make an overload for koffi.load to take Buffer as argument?

```ts
const dllBuffer = require('fs').readFileSync(dllPath);
const lib = koffi.load(dllBuffer);
```

Thanks a lot!

**Comment by Koromix** [comment]
I'd love to!

But almost no OS provides a way to do load a shared library from memory, except by writing it to a temporary file and loading it from there.

It is in theory possible to build an alternative DLL/so/dylib loader for each platform... But it's complicated, and it tends to fall short on modern OSes in many cases because for many reasons, including:
- Missing documentation for many corner cases
- Various security mitigations, which may require kernel involvement (W^X, PaX, Windows-specific mitigations, etc.)

For example, this project provides a way to load a DLL on Windows: https://github.com/fancycode/MemoryModule
But it can fail in many cases, or crash in obscure ways. In addition to being Windows only.

I don't want to add unreliable functionality to Koffi, unless Microsoft, Apple, and every Linux loader/libc decides to provide some kind of "LoadLibraryFromBuffer()" function. Unless I'm mistaken, as of now, this does not exist.

In short, you best bet it to write your buffer to a temporary file and use Koffi on this file. Sorry ><
